dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1

```diff
@@ -372,6 +375,74 @@ Configuration InstallPCoIPAgent
 			}
 		}
 
+		Script StartCAMIdleShutdownService
+        {
+            DependsOn  = @("[Script]StartPcoIPService")
+
+            GetScript  = { return 'Start CAM Idle Shutdown Service'}
+
+            TestScript = { 
+				$serviceName = "CAMIdleShutdown"
+				
+				if ([string]::IsNullOrWhiteSpace($using:enableAutoShutdown)) {
+					return $true
+				}
+
+				try {
+					$svc = Get-WmiObject -Class Win32_Service -Filter "Name='$ServiceName'"
+					return $svc.State -eq "Running"
+				} catch {
+					return $false
+				}
+            }
+
+            SetScript  = {
+				cd "C:\Program Files (x86)\Teradici\PCoIP Agent\bin\"
+				$serviceName = "CAMIdleShutdown"
+
+				$ret = .\IdleShutdownAgent.exe -install
+				# Check for success
+				if( !$? ) {
+					$msg = "Failed to install {0} because: {1}" -f $serviceName, $ret
+					Write-Verbose $msg
+					throw $msg
+				}
+
+				$svc = Get-Service -Name $serviceName   
+
+				if ($svc.StartType -ne "Automatic") {
+					$msg = "try setting {0} Service start type to automatic." -f $serviceName
+					Write-Verbose $msg
+
+					Set-Service -name  $serviceName -StartupType Automatic
```

Comment:
Similar comment to above -if 'Set-Service' has an error then even if you have error handling code after, DSC seems to sense the error (a disturbance in the force) and bork. This is a real PITA to test, naturally - did you see this handle these error conditions properly when you did your testing?